spring-kafka/src/main/java/org/springframework/kafka/support/AbstractKafkaHeaderMapper.java

```diff
@@ -191,6 +193,16 @@ public void addRawMappedHeader(String name, boolean toString) {
 		this.rawMappedHeaders.put(name, toString);
 	}
 
+	/**
+	 * Add patterns for matching multi-value headers under the same key.
+	 * @param patterns the patterns for header.
+	 */
+	public void addHeaderPatternsForMultiValue(String ... patterns) {
+		for (String pattern : patterns) {
+			this.headerMatchersForMultiValue.add(new SimplePatternBasedHeaderMatcher(pattern));
```

Comment:
I believe we can use Java `Stream` API and then `addAll` on the targer collection.